spanner/api/Spanner.Samples/UpdateDataAsync.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_update_data]
+
+using Google.Cloud.Spanner.Data;
+using System;
+using System.Threading.Tasks;
+
+public class UpdateDataAsyncSample
+{
+    public async Task<int> UpdateDataAsync(string projectId, string instanceId, string databaseId)
+    {
+        string connectionString = $"Data Source=projects/{projectId}/instances/{instanceId}/databases/{databaseId}";
+
+        using var connection = new SpannerConnection(connectionString);
+
+        var rowCount = 0;
+        SpannerCommand cmd = connection.CreateDmlCommand(
```

Comment:
The existing sample creates an update command instead of writing DML from scratch. Was there a reason to change it?

Reply:
There was suggestion by @amanda-tarafa 
@skuruppu Should this sample use UPDATE...WHERE? This is a very weird way of updating rows, i.e. while they are being read?

_Originally posted by @amanda-tarafa in https://github.com/GoogleCloudPlatform/dotnet-docs-samples/pull/1178#r496631153_

Reply:
Yes, I remember there was something unusual going on with one of the original samples. It was doing some reads to add some read values as parameters where it could just do UPDATE...WHERE.